crates/cli/src/cli.rs

```diff
@@ -551,6 +561,147 @@ impl Cli {
         Ok(config)
     }
 
+    pub fn into_telemetry_props(self) -> TelemetryProps {
+        TelemetryProps::new()
+            .insert("command", get_cli_command_telemetry_props(self.command))
+            .insert_with("offline", self.offline, |v| v.then_some(v))
+            .insert_with("health_check_endpoint", self.health_check_endpoint, |v| {
+                v.then_some(v)
+            })
+            .insert_with("config_out", self.config_out, |v| {
+                v.map(|_| TELEMETRY_SENSITIVE_VALUE)
+            })
+            .insert_with("port", self.port, |v| {
+                v.filter(|&p| p.to_string() != DEFAULT_PORT)
+                    .map(|_| TELEMETRY_SENSITIVE_VALUE)
+            })
+            .insert_with("host", &self.host, |v| {
+                v.first()
+                    .filter(|&h| h.to_string() != DEFAULT_HOST || self.host.len() != 1)
+                    .map(|_| TELEMETRY_SENSITIVE_VALUE)
+            })
+            .insert_with("chain_id", self.chain_id, |v| {
+                v.map(|_| TELEMETRY_SENSITIVE_VALUE)
+            })
+            .insert_with("debug_mode", self.debug_mode, |v| v.then_some(v))
+            .insert_with("show_node_config", self.show_node_config, |v| {
+                (!v.unwrap_or(false)).then_some(false)
+            })
+            .insert_with("show_tx_summary", self.show_tx_summary, |v| {
+                (!v.unwrap_or(false)).then_some(false)
+            })
+            .insert("disable_console_log", self.disable_console_log)
+            .insert("show_event_logs", self.show_event_logs)
+            .insert("show_calls", self.show_calls.map(|v| v.to_string()))
+            .insert("show_outputs", self.show_outputs)
+            .insert(
+                "show_storage_logs",
+                self.show_storage_logs.map(|v| v.to_string()),
+            )
+            .insert(
+                "show_vm_details",
+                self.show_vm_details.map(|v| v.to_string()),
+            )
+            .insert(
+                "show_gas_details",
+                self.show_gas_details.map(|v| v.to_string()),
+            )
+            .insert("resolve_hashes", self.resolve_hashes)
```

Comment:
Note: the flags `resolve_hashes`, `show_output`,  `show_event_logs`, `show_calls` will be removed in `0.4.0` and will be replaced with `verbosity levels` (e.g. -v, -vv, -vvv). 

Nothing to do here, but wanted to mention for awareness and likely changes needed when  https://github.com/matter-labs/anvil-zksync/pull/577 lands